src/services/mongoose.ts

```diff
@@ -212,4 +213,62 @@ export class Repository<T extends IModel> {
 			items
 		};
 	}
+
+	public async autoComplete(options: AutoCompleteOptions<T>) {
+		const skip = options.offset == undefined ? 0 : options.offset;
+		const limit = options.limit == undefined ? 100 : options.limit;
+		const pipeline: PipelineStage[] = [
+			{
+				$search: {
+					index: options.indexName,
+					autocomplete: {
+						path: options.path,
+						query: options.query,
+						fuzzy: {
+							// optional, but worth testing for bad spellers
+							maxEdits: 1, // up to 1 mispelled char
+							prefixLength: 2, // minimum correct letters before fixing
+							maxExpansions: 5 // max variations to search for
+						}
+					}
+				}
+			},
+			{
+				$skip: skip
+			},
+			{
+				$limit: limit
+			}
+		];
+		return this.aggregate<T[]>(pipeline);
+	}
+
+	public async textSearch(options: AutoCompleteOptions<T>) {
```

Comment:
This looks pretty much identical to the `autoComplete` function above - do we need both of them?